diff --git a/internal/services/dns_record/resource.go b/internal/services/dns_record/resource.go
index 577fadc..a19ba25 100755
--- a/internal/services/dns_record/resource.go
+++ b/internal/services/dns_record/resource.go
@@ -164,6 +164,11 @@ func (r *DNSRecordResource) Read(ctx context.Context, req resource.ReadRequest,
 		return
 	}

+	if data.ID.ValueString() == "" || data.ZoneID.ValueString() == "" {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
 	res := new(http.Response)
 	env := DNSRecordResultEnvelope{*data}
 	_, err := r.client.DNS.Records.Get(
diff --git a/internal/services/zone/resource.go b/internal/services/zone/resource.go
index 6151f0b..0e79bbc 100644
--- a/internal/services/zone/resource.go
+++ b/internal/services/zone/resource.go
@@ -150,6 +150,11 @@ func (r *ZoneResource) Read(ctx context.Context, req resource.ReadRequest, resp
 		return
 	}

+	if data.ID.ValueString() == "" {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+
 	res := new(http.Response)
 	env := ZoneResultEnvelope{*data}
 	_, err := r.client.Zones.Get(
